### PR TITLE
Appending Check And Message At the Batch Check API Response

### PR DESCRIPTION
**Pull request by daveaugustus**
### :nut_and_bolt: Description: What code changed, and why?
To Attach the response for Batch Check API, we need check and message from each API trigger.

For appending  this information, We can use Batch Check Service when it receives the response from all the API. We can attach Msg And Check type in the API result itself.  

### :+1: Definition of Done
- The check and message should be appended
- Test cases should be modified and passing

### :athletic_shoe: How to Build and Test the Change
Run the server using the following command and trigger the batchcheck API
`make verify-serve`
### :white_check_mark: Checklist

**All PRs** must tick these:

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [x] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [x] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [x] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [x] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [x] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [x] Tests added/updated? (all new code needs new tests)
- [x] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### :camera: Screenshots, if applicable
Screenshot 2023-06-01 at 10 09 15 AM

Screenshot 2023-06-01 at 5 13 12 PM

Screenshot 2023-06-02 at 2 10 40 PM

### CODE REVIEW

File: components/automate-cli/pkg/verifyserver/constants/checkconstant.go [hunk 1]
```diff
@@ -14,33 +18,121 @@ const AWS_OPENSEARCH_S3_BUCKET_ACCESS = "aws-opensearch-s3-bucket-access"
...
+)
 
 const NFS_MOUNT_MSG = "NFS Mount Backup Check"
 
+const (
+	BastionCheckType = "bastion"
+	RemoteCheckType  = "remote"
+)
+
+var abcMap map[string]models.CheckAndType = map[string]models.CheckAndType{
```
Comment (kalroy) [inline components/automate-cli/pkg/verifyserver/constants/checkconstant.go:44]: Name the variable properly. Give relevant name

File: components/automate-cli/pkg/verifyserver/services/batchcheckservice/batchcheckservice.go [hunk 2]
```diff
@@ -135,15 +139,21 @@ func constructBatchCheckResponse(checkTriggerRespMap map[string][]models.CheckTr
...
 	var resultIndex = 0
 	for _, v := range ipMap {
+		if len(v) == 0 {
+			continue
+		}
 		result[resultIndex].Ip = v[0].Host
 		result[resultIndex].NodeType = v[0].NodeType
 		resultArray := []models.ApiResult{}
 		for _, checkResult := range v {
+			checkResult.Result.Message = constants.GetCheckByType(checkResult.CheckType)
```
Comment (kalroy) [inline components/automate-cli/pkg/verifyserver/services/batchcheckservice/batchcheckservice.go:149]: Why can't we make this changes to the response in one place at L47?

File: components/automate-cli/pkg/verifyserver/constants/checkconstant.go [hunk 3]
```diff
@@ -14,33 +18,121 @@ const AWS_OPENSEARCH_S3_BUCKET_ACCESS = "aws-opensearch-s3-bucket-access"
...
+		CheckMsg:  EXTERNAL_POSTGRESQL_MSG,
+	},
+	NFS_BACKUP_CONFIG: {
+		CheckType: RemoteCheckType,
+		CheckName: NFS_BACKUP_CONFIG,
+		CheckMsg:  NFS_BACKUP_CONFIG_MSG,
+	},
+}
+
+func GetCheckByType(checkName string) string {
```
Comment (kalroy) [inline components/automate-cli/pkg/verifyserver/constants/checkconstant.go:112]: Give relevant function name.
```suggestion
func GetCheckMessageByType(checkType string) string {
```

**Review by kalroy — CHANGES_REQUESTED** [review]

File: components/automate-cli/pkg/verifyserver/constants/checkconstant.go [hunk 4]
```diff
@@ -14,33 +18,121 @@ const AWS_OPENSEARCH_S3_BUCKET_ACCESS = "aws-opensearch-s3-bucket-access"
...
+	EXTERNAL_OPENSEARCH_MSG             = "External Opensearch Check"
+	AWS_OPENSEARCH_S3_BUCKET_ACCESS_MSG = "AWS Opensearch S3 Bucket Access Check"
+	EXTERNAL_POSTGRESQL_MSG             = "External Postgresql Check"
+	NFS_BACKUP_CONFIG_MSG               = "Nfs Backup Config Check"
+)
 
 const NFS_MOUNT_MSG = "NFS Mount Backup Check"
 
+const (
+	BastionCheckType = "bastion"
```
Comment (YashviJain01) [inline components/automate-cli/pkg/verifyserver/constants/checkconstant.go:40]: Bastion

File: components/automate-cli/pkg/verifyserver/constants/checkconstant.go [hunk 5]
```diff
@@ -14,33 +18,121 @@ const AWS_OPENSEARCH_S3_BUCKET_ACCESS = "aws-opensearch-s3-bucket-access"
...
+	SYSTEM_RESOURCES_MSG                = "System Resources Check"
+	SOFTWARE_VERSIONS_MSG               = "Software Versions Check"
+	SYSTEM_USER_MSG                     = "System User Check"
+	S3_BACKUP_CONFIG_MSG                = "S3 Backup Config Check"
+	FQDN_MSG                            = "FQDN Check"
+	FIREWALL_MSG                        = "Firewall Check"
+	EXTERNAL_OPENSEARCH_MSG             = "External Opensearch Check"
+	AWS_OPENSEARCH_S3_BUCKET_ACCESS_MSG = "AWS Opensearch S3 Bucket Access Check"
+	EXTERNAL_POSTGRESQL_MSG             = "External Postgresql Check"
+	NFS_BACKUP_CONFIG_MSG               = "Nfs Backup Config Check"
```
Comment (YashviJain01) [inline components/automate-cli/pkg/verifyserver/constants/checkconstant.go:34]: Should this be NFS Backup Config Check

File: components/automate-cli/pkg/verifyserver/constants/checkconstant.go [hunk 6]
```diff
@@ -14,33 +18,121 @@ const AWS_OPENSEARCH_S3_BUCKET_ACCESS = "aws-opensearch-s3-bucket-access"
...
+	return val.CheckMsg
+}
+
 func GetBastionChecks() []string {
-	var checks = []string{
-		HARDWARE_RESOURCE_COUNT,
-		CERTIFICATE,
-		SSH_USER,
+	var checks []string
+	for _, v := range abcMap {
```
Comment (YashviJain01) [inline components/automate-cli/pkg/verifyserver/constants/checkconstant.go:122]: we can use the key also over here and append the keys directly.

File: components/automate-cli/pkg/verifyserver/constants/checkconstant_test.go [hunk 7]
```diff
@@ -9,22 +9,42 @@ import (
...
+		resp := GetCheckByType(NFS_BACKUP_CONFIG)
+		assert.Equal(t, resp, NFS_BACKUP_CONFIG_MSG)
+	})
+
+	t.Run("Get FIREWALL_MSG", func(t *testing.T) {
+		resp := GetCheckByType(FIREWALL)
+		assert.Equal(t, resp, FIREWALL_MSG)
+	})
+
+	t.Run("Get FIREWALL_MSG", func(t *testing.T) {
```
Comment (YashviJain01) [inline components/automate-cli/pkg/verifyserver/constants/checkconstant_test.go:44]: Rename the name for this, this is not a Firewall mesage check. And add test cases for other checks as well

File: components/automate-cli/pkg/verifyserver/services/batchcheckservice/batchcheckservice.go [hunk 8]
```diff
@@ -135,15 +139,21 @@ func constructBatchCheckResponse(checkTriggerRespMap map[string][]models.CheckTr
 	var result = make([]models.BatchCheckResult, len(ipMap))
 	var resultIndex = 0
 	for _, v := range ipMap {
+		if len(v) == 0 {
+			continue
+		}
 		result[resultIndex].Ip = v[0].Host
 		result[resultIndex].NodeType = v[0].NodeType
 		resultArray := []models.ApiResult{}
 		for _, checkResult := range v {
```
Comment (YashviJain01) [inline components/automate-cli/pkg/verifyserver/services/batchcheckservice/batchcheckservice.go:148]: Can we get the checktype from the response and append message accordingly ?

File: components/automate-cli/pkg/verifyserver/services/batchcheckservice/batchcheckservice.go (hunk 2 above)
Comment (daveaugustus) [reply components/automate-cli/pkg/verifyserver/services/batchcheckservice/batchcheckservice.go]: @kalroy if we do it there for remote check then we need to do it for bastion check also a block up in the same file. Hence I decided to keep where the response is being constructed.

File: components/automate-cli/pkg/verifyserver/services/batchcheckservice/batchcheckservice.go (hunk 8 above)
Comment (daveaugustus) [reply components/automate-cli/pkg/verifyserver/services/batchcheckservice/batchcheckservice.go:148]: Yes we are getting the check type in checkResult.CheckType.

File: components/automate-cli/pkg/verifyserver/services/batchcheckservice/batchcheckservice.go (hunk 2 above)
Comment (daveaugustus) [reply components/automate-cli/pkg/verifyserver/services/batchcheckservice/batchcheckservice.go]: @kalroy moved!

File: components/automate-cli/pkg/verifyserver/constants/checkconstant.go [hunk 9]
```diff
@@ -14,33 +18,121 @@ const AWS_OPENSEARCH_S3_BUCKET_ACCESS = "aws-opensearch-s3-bucket-access"
...
 const NFS_MOUNT_MSG = "NFS Mount Backup Check"
 
+const (
+	BASTION_CHECK = "bastion"
+	REMOTE_CHECK  = "remote"
+)
+
+var checkMap map[string]models.CheckAndType = map[string]models.CheckAndType{
+	HARDWARE_RESOURCE_COUNT: {
+		CheckName: BASTION_CHECK,
```
Comment (Dmaddu) [inline components/automate-cli/pkg/verifyserver/constants/checkconstant.go:46]: CheckType should hold the type of check i.e. Bastion or Remote
and CheckName should hold the name of the check. 
You swapped both the fields

**Review by Dmaddu — CHANGES_REQUESTED** [review]

File: components/automate-cli/pkg/verifyserver/constants/checkconstant.go [hunk 10]
```diff
@@ -14,33 +18,121 @@ const AWS_OPENSEARCH_S3_BUCKET_ACCESS = "aws-opensearch-s3-bucket-access"
...
+		CheckMsg:  EXTERNAL_POSTGRESQL_MSG,
+	},
+	NFS_BACKUP_CONFIG: {
+		CheckType: REMOTE_CHECK,
+		CheckName: NFS_BACKUP_CONFIG,
+		CheckMsg:  NFS_BACKUP_CONFIG_MSG,
+	},
+}
+
+func GetCheckMessageByType(CheckName string) string {
```
Comment (Dmaddu) [inline components/automate-cli/pkg/verifyserver/constants/checkconstant.go:112]: This should be GetCheckMessageByName

File: components/automate-cli/pkg/verifyserver/services/batchcheckservice/batchcheckservice.go [hunk 11]
```diff
@@ -144,8 +149,49 @@ func constructBatchCheckResponse(checkTriggerRespMap map[string][]models.CheckTr
...
+	for _, check := range bastionChecks {
+		go ss.RunBastionCheck(check, config, bastionCheckResultChan)
+	}
+
+	// iterate over the chan and take the value out and populate checkTriggerRespMap
+	for i := 0; i < len(bastionChecks); i++ {
+		result := <-bastionCheckResultChan
+		for i, _ := range result {
+			result[i].Result.Check = result[i].CheckType
+			result[i].Result.Message = constants.GetCheckMessageByType(result[i].Result.Check)
```
Comment (Dmaddu) [inline components/automate-cli/pkg/verifyserver/services/batchcheckservice/batchcheckservice.go:170]: You missed this one

**Review by Dmaddu — CHANGES_REQUESTED** [review]

**Review by Dmaddu — APPROVED** [review]

**Review by SahithiMy — APPROVED** [review]

**Review by YashviJain01 — APPROVED** [review]

**Pull request merged**